### PR TITLE
Use `fori_loop`

### PR DESCRIPTION
**Pull request by Smit-create**
#112

### CODE REVIEW

File: lectures/inventory_dynamics.md [hunk 1]
```diff
@@ -375,26 +372,23 @@ def compute_freq(firm, key,
...
-        restock_count = jnp.where(X <= s,
-                            restock_count + 1,
-                            restock_count)
-        X = jnp.where(X <= s, 
-                      jnp.maximum(S - D, 0),
-                      jnp.maximum(X - D, 0))
-        
-        Xs = jnp.vstack((X, restock_count))
-        return Xs, None
+        x, restock_cnt = Xs[0], Xs[1]
```
Comment (jstac) [inline lectures/inventory_dynamics.md:377]: please change back to `restock_count`.

File: lectures/inventory_dynamics.md [hunk 2]
```diff
@@ -375,26 +372,23 @@ def compute_freq(firm, key,
...
+        restock_cnt = jnp.where(x <= s,
+                                restock_cnt + 1,
+                                restock_cnt)
+        x = jnp.where(x <= s,
+                      jnp.maximum(S - D[i], 0),
+                      jnp.maximum(x - D[i], 0))
 
-    # Use lax.scan to perform the calculations on all states
-    X_final, _ = lax.scan(update_X, Xs, D)
+        Xs = jnp.vstack((x, restock_cnt))
```
Comment (jstac) [inline lectures/inventory_dynamics.md:385]: This is a comment about the existing code by @HumphreyYang , but why are we using `vstack` here?  Can't we just use a tuple?

**Comment by jstac** [comment]
Thanks @Smit-create , this is looking good.  Just small comments and requests above.

**Comment by HumphreyYang** [comment]
Many thanks @jstac and @Smit-create,

I have updated `np.vstack` and the request on reverting the change on `restock_count` as I went through the code.

**Comment by jstac** [comment]
Thanks @HumphreyYang , much appreciated.

**Comment by Smit-create** [comment]
Thank you @HumphreyYang  and @jstac.

**Pull request merged**